lib/Cake/TestSuite/CakeTestCase.php

```diff
@@ -68,52 +73,6 @@ abstract class CakeTestCase extends \PHPUnit\Framework\TestCase {
  */
 	protected $_pathRestore = array();
 
-/**
- * Runs the test case and collects the results in a TestResult object.
- * If no TestResult object is passed a new one will be created.
- * This method is run for each test method in this class
- *
- * @param TestResult $result The test result object
- * @return TestResult
- * @throws InvalidArgumentException
- */
-	public function run(TestResult $result = null): TestResult {
-		$level = ob_get_level();
-
-		if (!empty($this->fixtureManager)) {
-			$this->fixtureManager->load($this);
-		}
-		$result = parent::run($result);
-		if (!empty($this->fixtureManager)) {
-			$this->fixtureManager->unload($this);
-			unset($this->fixtureManager, $this->db);
-		}
-
-		for ($i = ob_get_level(); $i < $level; ++$i) {
-			ob_start();
-		}
-
-		return $result;
-	}
-
-/**
- * Called when a test case method is about to start (to be overridden when needed.)
- *
- * @param string $method Test method about to get executed.
- * @return void
- */
-	public function startTest($method) {
-	}
-
-/**
- * Called when a test case method has been executed (to be overridden when needed.)
- *
- * @param string $method Test method about that was executed.
- * @return void
- */
-	public function endTest($method) {
```

Comment:
These methods did literally nothing. They're not overriding a parent method or anything.

Reply:
They are there to be overridden aren't they? Like abstract methods but they don't _need_ to be implemented.